lib/practice.md

```diff
@@ -3,58 +3,74 @@
 - What package did you install for enumeration?
 
 ```md
-<!-- your answer here -->
+lodash
 ```
 
 ## Research: Popular NPM Packages
 
 Package 1:
 ```md
-<!-- your answer here -->
+async
 ```
 1.  What problem does this package solve?
 ```md
-<!-- your answer here -->
+asynchronous control flow
 ```
 1.  How can I use this package in my own code? What steps are involved?
 ```md
-<!-- your answer here -->
+`npm install --save async`
+
+require async in the file where I want to use it
+
+cry myself to sleep because I hate this library so much
 ```
 1.  How well-maintained is this package? What criteria did I use to decide?
 ```md
-<!-- your answer here -->
+seems pretty well maintained, based on frequency of commits and number of active and closed issues/pull requests.
 ```
 
 Package 2:
 ```md
-<!-- your answer here -->
+underscore
 ```
 1.  What problem does this package solve?
 ```md
-<!-- your answer here -->
+I honestly do not know what problem this solves. Probably JavaScript devs not having enough random dependencies.
 ```
 1.  How can I use this package in my own code? What steps are involved?
 ```md
-<!-- your answer here -->
+I guess I would use it to help with functional programming problems that I don't know I have yet?
+
+update: I talked to a friend of mine who is doing a PhD in computer science and she said that underscore turns JavaScript into a much more useable functional language.
+
+This actually leads me to a question: Is it fair to say that naked JavaScript is stuck halfway between being a functional language and an object oriented one?
```

Comment:
I'm not quite sure which programming paradigm JavaScript fits into.  My impression is that it has some, but not all, properties from both functional and object-oriented programming languages.  The big difference, from my experience, is JavaScript using prototypal inheritance over classical inheritance.  It's possible to emulate (not sure if this is the correct term) classical inheritance in JavaScript using the prototype chain.  All data in JavaScript is also public unless a closure is used to hide it, while languages like Java and C# have visibility modifiers (public, private, protected, etc.) for fields, methods, classes, etc.  This article ([Comparison of programming paradigms](https://en.wikipedia.org/wiki/Comparison_of_programming_paradigms)) might of interest to you.  I wish that I were able to give you a better answer to this question.  You should consider asking Antony this question during his "Ask Me Anything" session since he can give you a much better answer.